projects/ngx-avatar/src/lib/sources/async-source.ts

```diff
@@ -8,8 +8,8 @@ import { AvatarSource } from './avatar-source.enum';
 export abstract class AsyncSource implements Source {
   readonly sourceType: AvatarSource;
 
-  constructor(public sourceId: string) {}
+  protected constructor(public sourceId: string) {}
```

Comment:
What's the reason for protecting this?

Reply:
The class is abstract and thus a public constructor cannot be used. Technically it doesn't change much, but it's to clarify intent.

See: https://stackoverflow.com/a/761864/37706

Reply:
Won't this make inheritance more difficult?

Reply:
Actually you are correct. Making the constructor protected, makes it impossible to have a inheriting class without constructor at all (and thus relying on the parent constructor). As demonstrated in https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=9&pc=24#code/IYIwzgLgTsDGEAJYBthjA4CDeAoBBCADlAPYQCm8FAJkqQHaRQCu8pUAFESyMgJaxMwAFwJm-BgHMAlDnyEAvrmW4UaDCAQUAHpQY0MWPKtiNICAI4IAvAgYUA7ghCcARGhpuZAbiA

I'll revert this.

Reply:
No problem.